### PR TITLE
feat: vesting managed

### PR DESCRIPTION
**Pull request by oopcode**
This PR adds a version of the vesting contract that allows the owner to delete existing vesting schedules for a given user.

The only addition to the astroport base besting contract is the `remove_vesting_accounts()` function: https://github.com/neutron-org/neutron-tge-contracts/blob/08f19f0e0c676000a7608369abc961c979c803b2/contracts/vesting-managed/src/contract.rs#L131

For each vesting account, we calculate the amount of tokens to claw back (unclaimed + still vesting), remove the vesting information from the storage, and update the `total_granted`, `total_released` metrics.

No neutron integration tests were implemented because I was able to write "integration" tests in the unit tests: https://github.com/neutron-org/neutron-tge-contracts/blob/aa9e1649cd1e4bb8459dad95282d82495e264603/contracts/vesting-managed/src/tests/integration.rs#L885

### CODE REVIEW

**Review by pr0n00gler — APPROVED** [review]
Looks legit

**Review by foxpy — COMMENTED** [review]
I think a README.md for vesting-managed wouldn't hurt.

File: contracts/vesting-managed/src/contract.rs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+///
+/// * **ExecuteMsg::Receive(msg)** Receives a message of type [`Cw20ReceiveMsg`] and processes it
+/// depending on the received template.
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let vest_app = BaseVesting::new(Strategy::Never);
```
[the remark this answers is not in the document]
Comment (oopcode) [reply contracts/vesting-managed/src/contract.rs:51]: no, this one doesn't need it!

File: contracts/vesting-managed/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+[package]
+name = "vesting-managed"
```
[the remark this answers is not in the document]
Comment (oopcode) [reply contracts/vesting-managed/Cargo.toml:2]: wow! i guess it got wiped out by one of the merges

File: contracts/vesting-managed/src/contract.rs [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                // Here we choose the "forget about everything" strategy. E.g., if we granted a user
+                // 300 tokens, and they claimed 150 tokens, the vesting state is
+                // { total_granted: 300, total_released: 150 }.
+                // If after that we remove the user's vesting account, we set the vesting state to
+                // { total_granted: 0, total_released: 0 }.
+                //
+                // If we decided to set it to { total_granted: 150, total_released: 150 }., the
+                // .total_released value of the vesting state would not be equal to the sum of the
+                // .released_amount values of all registered accounts.
+                let mut state = s.ok_or(ContractError::AmountIsNotAvailable {})?;
```
[the remark this answers is not in the document]
Comment (oopcode) [reply contracts/vesting-managed/src/contract.rs:183]: no, this is exactly what is done in the base code, and i want to keep it uniform

File: contracts/vesting-managed/src/contract.rs [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    // For each vesting account, calculate the amount of tokens to claw back (unclaimed + still
+    // vesting), transfer the required amount to the owner, remove the vesting information
+    // from the storage, and decrease the total granted metric.
+    for vesting_account in vesting_accounts {
+        let mut total_granted_for_user = Uint128::zero();
+
+        let account_address = deps.api.addr_validate(&vesting_account)?;
+        if let Some(account_info) = vesting_info.may_load(deps.storage, &account_address)? {
+            for sch in account_info.schedules {
+                if let Some(end_point) = sch.end_point {
```
[the remark this answers is not in the document]
Comment (oopcode) [reply contracts/vesting-managed/src/contract.rs:158]: fixed!

[the remark this answers is not in the document]
Comment (sotnikov-s) [reply contracts/vesting-managed/src/contract.rs:158]: why don't you want to use the func I gave you a link to? it does just the same as your part of code at the lines 156-168

[the remark this answers is not in the document]
Comment (oopcode) [reply contracts/vesting-managed/src/contract.rs:158]: wrong func?

**Pull request merged**